src/wasm.h

```diff
@@ -1427,15 +1427,15 @@ class Function {
   }
 };
 
+enum class Kind {
```

Comment:
Maybe `GlobalKind`? What you are describing sounds a lot like LLVM's `GlobalValue` (from which `GlobalVariable` and `Function` are derived). 
Actually, do `Table` and `Memory` fall into the same taxonomy for `GlobalOpt`-like purposes? If not, then maybe we don't want to reuse the same enum for that.